react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,124 @@
+### Introduction
+
+In this lesson we will cover keys in React. Keys are special props for our components and we'll learn why they are used.
+
+### Lesson overview
+
+By the end of the lesson you should be able to answer the following:
+
+- What are keys and why does React need them?
+- What are good and bad examples of keys?
+
+### Why does React need keys?
+
+In the upcoming lessons as you learn more about the internal workings of React, more specifically the rerendering process, you will understand the importance of keys. For now we will keep it simple. Recall the previous lesson on rendering lists. We used the `.map()` method to iterate over an array of data and return a list of elements. Now imagine, if any of the item in the list were to change, how would React know which item to update?
```

Comment:
```suggestion
In the upcoming lessons as you learn more about the internal workings of React, more specifically the rerendering process, you will understand the importance of keys. For now we will keep it simple. 

In the previous lesson on rendering lists we used the `.map()` method to iterate over an array of data and return a list of elements. Now imagine, if any of the item in the list were to change, how would React know which item to update?
```

Do you think this sounds better? I felt like this needed to be split out more and I changed it a bit because the sentence felt a bit chopped up. What do you think?

Reply:
I like this.